### PR TITLE
controlplane developer improvements

### PR DESCRIPTION
**Pull request by aanm**
These commits will help the update process of new k8s versions that we support. Please read on a per commit basis. Feedback welcome.

### CODE REVIEW

**Comment by aanm** [comment]
travis hit https://github.com/cilium/cilium/issues/21145, I've re-run the test again.

File: test/controlplane/k8s_versions.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    case $arg in
+        "--update-kind-config" )
+            update-kind-config
+        ;;
+        "--pre-pull-images" )
+            pre-pull-images
+        ;;
+   esac
+done
+>>>>>>> 6949c3e931 (pre-pull-docker-images)
```
Comment (sayboras) [inline test/controlplane/k8s_versions.sh]: this line seems unintentional.

File: test/controlplane/Makefile [hunk 2]
```diff
@@ -7,3 +7,15 @@ test:
...
+pre-pull-kind-images:
+	bash ./k8s_versions.sh --pre-pull-images
+
+update-k8s-versions:
+	# Delete all auto-generated input test files
+	find . -type d -regextype posix-extended -regex ".*/v[0-9]{1}\.[0-9]{2}" | xargs rm -r || true
+	# Update kind-config with the versions stored in k8s_versions.txt
+	$(shell bash ./k8s_versions.sh --update-kind-config)
+
+generate-input-files: pre-pull-kind-images update-k8s-versions update-golden test
```
Comment (sayboras) [inline test/controlplane/Makefile]: not sure if Makefile target is honouring the order of execution, I ran `make update-k8s-versions generate-input-files` locally to test it out, but got below error 

```
    testcase.go:252: Failed to read /home/tammach/go/src/github.com/cilium/cilium/test/controlplane/services/nodeport/v1.23/init.yaml: open /home/tammach/go/src/github.com/cilium/cilium/test/controlplane/services/nodeport/v1.23/init.yaml: no such file or directory
....
```

File: test/controlplane/Makefile [hunk 3]
```diff
@@ -7,3 +7,15 @@ test:
 
 update-golden:
 	go test . -test.v -update
+
+pre-pull-kind-images:
+	bash ./k8s_versions.sh --pre-pull-images
+
+update-k8s-versions:
+	# Delete all auto-generated input test files
+	find . -type d -regextype posix-extended -regex ".*/v[0-9]{1}\.[0-9]{2}" | xargs rm -r || true
```
Comment (sayboras) [inline test/controlplane/Makefile]: any reason to remove files here? is it mainly to clean up old version?

I am thinking about only overwrite existing file but not remove, we can only remove if the kube version is not longer mentioned in k8s_versions.txt.

**Review by joamaki — APPROVED** [review]
:+1:

File: test/controlplane/Makefile (hunk 3 above)
Comment (aanm) [reply test/controlplane/Makefile]: Great idea.

File: test/controlplane/Makefile (hunk 2 above)
Comment (aanm) [reply test/controlplane/Makefile]: oh yeah, the `update-golden test` should not be there.

**Review by sayboras — APPROVED** [review]
LGTM :100:

**Pull request merged**